src/commands/reactchannel.ts

```diff
@@ -0,0 +1,188 @@
+import { EmojiService } from '../services/EmojiService';
+import { Logger } from '../services/Logger';
+import { ReactService } from '../services/ReactService';
+import { ErrorMessageService } from '../services/ErrorMessageService';
+import { Collection, GuildChannel, Message, TextChannel } from 'discord.js';
+
+@Singleton
+@AutoWired
+export class ReactChannelCommand implements ICommand {
+
+  help = 'React to the last message in a given channel with a specific emoji\n' +
+    '**Usage**\n' +
+    'reactchannel channel-name-or-id emoji-name';
+  aliases = ['reactchannel', 'rc'];
+
+  @Inject private logger: Logger;
+  @Inject private emojiService: EmojiService;
+  @Inject private reactService: ReactService;
+  @Inject private errorMessageService: ErrorMessageService;
+
+  async execute(cmdArgs: ICommandArgs): Promise<ICommandResult> {
+    const {message, args} = cmdArgs;
+
+    // Can't apply a reaction to a channel when in a DM
+    if (message.channel.type !== 'text') {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  You cannot run this command in this chat!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Invalid message channel type', result: {success: false}};
+    }
+
+    const commandArgs = args.split(' ');
+
+    // Args should only be length 2
+    if (commandArgs.length !== 2) {
+      // Error, number of arguments doesn't match
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  You must include a channel name or id and an emoji name!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Argument error occurred', result: {success: false}};
+    }
+
+    const commandChannel = message.channel as TextChannel;
+
+    const channelIdOrName: string = commandArgs[0];
+    const emojiName: string = commandArgs[1];
+
+    if (commandChannel.id === channelIdOrName ||
+      commandChannel.name === channelIdOrName) {
+      return this.reactNow(emojiName, message);
+    }
+
+    const emoji = this.emojiService.getEmojiByName(emojiName);
+
+    if (!emoji) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find that emoji!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return { resultString: 'Could not find the emoji', result: {success: false}};
+    }
+
+    let channels: Collection<string, GuildChannel> = message.guild.channels;
+
+    if (!channels || !(channels.array()) || channels.array().length < 1) {
+      return this.couldNotFindTheChannel(message);
+    }
+
+    let textChannels: Collection<string, GuildChannel> = channels.filter((c) => c.type === 'text');
+
+    let channel: TextChannel = textChannels.find('id', channelIdOrName) as TextChannel;
+
+    if (!channel) {
+      channel = textChannels.find('name', channelIdOrName) as TextChannel;
```

Comment:
Could this be combined into a single || assignment? Or would that be unreadable / not valid?

let channel: TextChannel = textChannels.find('id', channelIdOrName) as TextChannel or textChannels.find('name', channelIdOrName) as TextChannel;